oodeel/methods/dknn.py

```diff
@@ -38,21 +38,28 @@ class DKNN(OODBaseDetector):
     Args:
         nearest: number of nearest neighbors to consider.
             Defaults to 1.
+        use_gpu (bool): Flag to enable GPU acceleration for FAISS. Defaults to False.
     """
 
-    def __init__(
-        self,
-        nearest: int = 50,
-    ):
+    def __init__(self, nearest: int = 50, use_gpu: bool = False):
         super().__init__()
-
         self.index = None
         self.nearest = nearest
+        self.use_gpu = use_gpu
+
+        if self.use_gpu:
+            try:
+                self.res = faiss.StandardGpuResources()
+            except AttributeError as e:
+                raise ImportError(
+                    "faiss-gpu is not installed, but use_gpu was set to True."
+                    + "Please install faiss-gpu or set use_gpu to False."
+                ) from e
```

Comment:
It is a little bit weird to check for the installation of a package within the code. It should be handled by the setup.cfg and automatically downloaded with pip.